app/metadata/metadata_store.py

```diff
@@ -76,16 +81,20 @@ def get_form_type(self):
         return self.get(MetaDataConstants.FORM_TYPE)
 
     def get_ref_p_start_date(self):
-        return self.get(MetaDataConstants.REF_P_START_DATE)
+        if self.user is None:
+            return None
+        return datetime.strptime(self.get(MetaDataConstants.REF_P_START_DATE), "%Y-%m-%d")
```

Comment:
I'd like to understand in what situations self.user is None as it shouldn't be.

Also you're already doing a self.user check in  self.get. I wonder if this would be better written:

def get_ref_p_start_date(self):
        start_date = self.get(MetaDataConstants.REF_P_START_DATE)		
        if start_date:
            return datetime.strptime(start_date, "%Y-%m-%d")